vocode/streaming/telephony/server/router/calls.py

```diff
@@ -55,6 +55,10 @@ def _from_call_config(
         synthesizer_factory: SynthesizerFactory = SynthesizerFactory(),
         events_manager: Optional[EventsManager] = None,
     ):
+        # if agent_factory.await_agent:
+        # Why is this better
```

Comment:
> why is this better

my understanding of `await_agent` was "use a flag to simulate an instanceof check"
* instanceof checks are already a bit of a code smell, and simulating one is worse IMO
* i was not able to understand this behavior without talking to you about it. Future engineers might not be able to do that

in comparison, calling a function on the factory class is _normal._ It's not doing anything unconventional and I think an intern could understand it at first glance

Reply:
also can you remove the code comment before merging

Reply:
instance checks are normal in python though, especially if you have different behaviour for certain subtypes. I feel like the new method is more implicit and obscures what is going on here more than before, an isintance would be best, but this is probably better than before. Another benefit of checking isinstance is that you can goto definition and it takes you directly to the function implementing it not the interface.
I understand that it is a code smell, but the code was structured polymorphically, and this is the simplest way todo what we want, without refactoring everything.

Reply:
Another way to view it is what I want is more like a discriminated union instead of straight up polymorphic interface, where we treat everything the same.